api/src/main/java/org/apache/iceberg/util/CharSequenceMap.java

```diff
@@ -41,7 +41,10 @@
  * <p>Note: This map does not support null keys.
  *
  * @param <V> the type of values
+ * @deprecated since 1.8.0, will be removed in 1.9.0. This class does not implement the Map contract
```

Comment:
Everything under the `api/` module requires a major version:
```suggestion
 * @deprecated since 1.8.0, will be removed in 2.0.0. This class does not implement the Map contract
```

Reply:
thanks, fixed!